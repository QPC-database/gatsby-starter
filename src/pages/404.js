import React from 'react'
import Footer from '../components/footer'
import HtmlHead from '../components/htmlhead'
import Layout from '../components/layout.js'
import { Link } from 'gatsby'

export const FourOhFourMarkup = props => (
	<Layout>
		<HtmlHead title="Ooops." />
		<header>
			<h1>Page not found</h1>
		</header>
		<main>
			<p>You might find what you're looking for at <Link to="/">the index page</Link>.</p>
		</main>
	</Layout>
)

export default FourOhFourMarkup
