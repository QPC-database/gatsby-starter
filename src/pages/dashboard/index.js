import { Router } from '@reach/router'
import HtmlHead from '../../components/htmlhead'
import Layout from '../../components/layout'
import React from 'react'

const Profile = () => (
	<>
		<h1>Your profile</h1>
		<ul>
			<li>You</li>
			<li>are you.</li>
		</ul>
	</>
)
const Login = () => (
	<>
		<h1>Login</h1>
		<ul>
			<li>Who</li>
			<li>are you?</li>
		</ul>
	</>
)

export class Dashboard extends React.Component {
	render() {
		const { children } = this.props
		return (
			<Layout>
				<HtmlHead title="Dashboard" />
				<Router>
					<Profile path='/dashboard/profile' />
					<Login path='/dashboard/login' />
				</Router>
			</Layout>
		)
	}
}

export default Dashboard
