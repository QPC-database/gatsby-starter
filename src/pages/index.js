import React from 'react'
import Bio from '../components/bio'
import HtmlHead from '../components/htmlhead'
import Layout from '../components/layout'

export const BodyMarkup = ({message}) => (
	<article>
		<h1>{message}</h1>
		<Bio />
	</article>
)

export const IndexMarkup = props => (
<Layout>
	<HtmlHead />
	<BodyMarkup message='Hello world!' />
</Layout>	
)

export default IndexMarkup
