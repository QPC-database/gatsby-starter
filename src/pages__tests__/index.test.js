import React from 'react'
import renderer from 'react-test-renderer'
import {BodyMarkup as HomePage} from '../pages/index'

describe(`HomePage`, () =>
  it(`renders correctly`, () => {
    const tree = renderer.create(<HomePage message='Hello world!' />).toJSON()
    expect(tree).toMatchSnapshot()
  }),
)
