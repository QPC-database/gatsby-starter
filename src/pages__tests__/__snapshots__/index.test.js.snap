// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<article>
  <h1>
    Hello world!
  </h1>
  <p
    data-testid="my-bio"
  >
    It is nice to meet you!  I am Gatsby.
  </p>
</article>
`;
